### ISSUE TITLE
An error occurred while Layout/RescueEnsureAlignment cop was inspecting in 0.71.0 against Ruby 2.6.1

### ISSUE THREAD

**Issue by jakeonrails** [rubocop-hq/rubocop#7099]
Just got this bug after upgrading to 0.71.0, but a quick search through issues shows this is happening for a while now

--------

## Expected behavior

No bug when running rubocop against this code:

```ruby
# frozen_string_literal: true

module Test
  X = lambda do
    x = 1
      rescue StandardError # should be indented only two spaces
    x
  end
end
```

## Actual behavior

```
Inspecting 1 file
An error occurred while Layout/RescueEnsureAlignment cop was inspecting /Users/jakemoffatt/Code/rubocop-dupe-bug/csv.rb:6:3.
To see the complete backtrace run rubocop -d.
C

Offenses:

csv.rb:7:4: C: Layout/IndentationWidth: Use 2 (not 1) spaces for indentation.
    x
   ^

1 file inspected, 1 offense detected

1 error occurred:
An error occurred while Layout/RescueEnsureAlignment cop was inspecting /Users/jakemoffatt/Code/rubocop-dupe-bug/csv.rb:6:3.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:
0.71.0 (using Parser 2.6.0.0, running on ruby 2.6.1 x86_64-darwin18)
```

## Steps to reproduce the problem

Run rubocop with the default configuration against the code sample above.

## RuboCop version

```
0.71.0 (using Parser 2.6.0.0, running on ruby 2.6.1 x86_64-darwin18)
```

**Comment by malyshkosergey** [comment]
@jakeonrails, I was looking at this issue and found the reason of the failure.
But I am not sure why rescue should be aligned two more spaces.

Current valid indentation is the following
```
module Test
          X = lambda do
            x = 1
          rescue StandardError
            x
          end
        end
```

Any objection about it?

**Comment by jakeonrails** [comment]
@malyshkosergey This is what I would expect the formatter to do and how I correct indentation manually. (Well, I would actually do this:

```ruby
module Test
  X = lambda do
    x = 1
  rescue StandardError
    x
  end
end
```

Basically I think the rescue should probably be aligned with the matching `end` token.

**Comment by jakeonrails** [comment]
Thank you for the fix @koic and @tatsuyafw :D
